test/OrchardCore.Tests/Localization/CultureScopeTests.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Globalization;
+using OrchardCore.Localization;
+using Xunit;
+
+namespace OrchardCore.Tests.Localization
+{
+    public class CultureScopeTests
+    {
+        [Fact]
+        public void CultureScopeSetUICultureAutomaticallyIfNotSet()
+        {
+            // Arrange
+            var culture = "ar-YE";
+
+            // Act
+            using (var cultureScope = CultureScope.Create(culture))
+            {
+                // Assert
+                Assert.Equal(culture, cultureScope.Culture.Name);
+                Assert.Equal(culture, cultureScope.UICulture.Name);
+            }
+        }
+
+        [Fact]
+        public void CultureScopeRetreivesBothCultureAndUICulture()
+        {
+            // Arrange
+            var culture = "ar";
+            var uiCulture = "ar-YE";
+
+            // Act
+            using (var cultureScope = CultureScope.Create(culture, uiCulture))
+            {
+                // Assert
+                Assert.Equal(culture, cultureScope.Culture.Name);
+                Assert.Equal(uiCulture, cultureScope.UICulture.Name);
+            }
+        }
+
+        [Fact]
+        public void CultureScopeRetreivesTheOrginalCulturesAfterScopeEnded()
+        {
+            // Arrange
+            var culture = CultureInfo.CurrentCulture;
+            var uiCulture = CultureInfo.CurrentUICulture;
+
+            // Act
+            using (var cultureScope = CultureScope.Create("FR"))
+            {
+
+            }
+
+            // Assert
+            Assert.Equal(culture, CultureInfo.CurrentCulture);
+            Assert.Equal(uiCulture, CultureInfo.CurrentUICulture);
+        }
+
+        [Fact]
+        public void CultureScopeRetreivesTheOrginalCulturesIfExceptionOccur()
```

Comment:
> E.g. when using this scope for liquid rendering (when not in a real view context)

@jtkech no need to test a fake `ILiquidTemplateManager` this unit tests ensures if the `Render` throws an exception it will restore the original cultures